cmd/kubeadm/app/cmd/phases/join/kubelet.go

```diff
@@ -150,44 +165,57 @@ func runKubeletStartJoinPhase(c workflow.RunData) (returnErr error) {
+	if err := kubeletphase.WriteKubeletDynamicEnvFile(&initCfg.ClusterConfiguration, &initCfg.NodeRegistration, registerTaintsUsingFlags, data.KubeletDir()); err != nil {
 		return err
 	}
 
 	// Try to start the kubelet service in case it's inactive
-	fmt.Println("[kubelet-start] Starting the kubelet")
-	kubeletphase.TryStartKubelet()
-
-	// Now the kubelet will perform the TLS Bootstrap, transforming /etc/kubernetes/bootstrap-kubelet.conf to /etc/kubernetes/kubelet.conf
-	// Wait for the kubelet to create the /etc/kubernetes/kubelet.conf kubeconfig file. If this process
-	// times out, display a somewhat user-friendly message.
-	waiter := apiclient.NewKubeWaiter(nil, kubeadmconstants.TLSBootstrapTimeout, os.Stdout)
-	if err := waiter.WaitForKubeletAndFunc(waitForTLSBootstrappedClient); err != nil {
-		fmt.Printf(kubeadmJoinFailMsg, err)
-		return err
-	}
+	if !data.DryRun() {
+		fmt.Println("[kubelet-start] Starting the kubelet")
+		kubeletphase.TryStartKubelet()
 
-	// When we know the /etc/kubernetes/kubelet.conf file is available, get the client
-	client, err := kubeconfigutil.ClientSetFromFile(kubeadmconstants.GetKubeletKubeConfigPath())
-	if err != nil {
-		return err
+		// Now the kubelet will perform the TLS Bootstrap, transforming /etc/kubernetes/bootstrap-kubelet.conf to /etc/kubernetes/kubelet.conf
+		// Wait for the kubelet to create the /etc/kubernetes/kubelet.conf kubeconfig file. If this process
+		// times out, display a somewhat user-friendly message.
+		waiter := apiclient.NewKubeWaiter(nil, kubeadmconstants.TLSBootstrapTimeout, os.Stdout)
+		if err := waiter.WaitForKubeletAndFunc(waitForTLSBootstrappedClient); err != nil {
+			fmt.Printf(kubeadmJoinFailMsg, err)
+			return err
+		}
+
+		// When we know the /etc/kubernetes/kubelet.conf file is available, get the client
+		client, err := kubeconfigutil.ClientSetFromFile(kubeadmconstants.GetKubeletKubeConfigPath())
+		if err != nil {
+			return err
+		}
+
+		klog.V(1).Infoln("[kubelet-start] preserving the crisocket information for the node")
+		if err := patchnodephase.AnnotateCRISocket(client, cfg.NodeRegistration.Name, cfg.NodeRegistration.CRISocket); err != nil {
+			return errors.Wrap(err, "error uploading crisocket")
+		}
 	}
 
-	klog.V(1).Infoln("[kubelet-start] preserving the crisocket information for the node")
-	if err := patchnodephase.AnnotateCRISocket(client, cfg.NodeRegistration.Name, cfg.NodeRegistration.CRISocket); err != nil {
-		return errors.Wrap(err, "error uploading crisocket")
+	// If we're dry-running, print the kubelet config manifests and print static pod manifests if joining a control plane
+	joinAsControlPlane := false
+	if cfg.ControlPlane != nil {
+		joinAsControlPlane = true
+	}
+	if err := printFilesIfDryRunning(data, joinAsControlPlane); err != nil {
+		return errors.Wrap(err, "error printing files on dryrun")
```

Comment:
```suggestion
	// If we're dry-running, print the kubelet config manifests and print static pod manifests if joining a control plane
	if err := printFilesIfDryRunning(data, cfg.ControlPlane != nil); err != nil {
		return errors.Wrap(err, "error printing files on dryrun")
```

Reply:
need to simplify here, fixed, thanks a lot